### PR TITLE
feat: add rotateAxisAngle for Quaternion

### PR DESCRIPTION
**Pull request by JujieX**
### Please check if the PR fulfills these requirements

- [ ] The commit message follows our [guidelines](https://github.com/oasis-engine/engine/blob/main/.github/COMMIT_MESSAGE_CONVENTION.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)
feat: add rotateAxisAngle for Quaternion
### What is the current behavior? (You can also link to an open issue here)
no such calculation
### What is the new behavior (if this is a feature change)?
rotateAxisAngle 
### Other information:

### CODE REVIEW

File: packages/math/src/Quaternion.ts [hunk 1]
```diff
@@ -697,4 +698,16 @@ export class Quaternion implements IClone {
...
   }
+
+  /**
+   * Create this quaternion rotate around an arbitrary axis.
+   * @param axis - The axis
+   * @param rad - The rotation angle in radians
+   * @returns This quaternion
+   */
+  rotateAxisAngle(axis: Vector3, rad: number): Quaternion {
+    Quaternion._tempQuat1.rotateAxisAngle(axis, rad);
```
Comment (gz65555) [inline packages/math/src/Quaternion.ts:709]: `Quaternion.rotateAxisAngle` is dead loop.

File: packages/math/src/Quaternion.ts [hunk 2]
```diff
@@ -697,4 +698,16 @@ export class Quaternion implements IClone {
...
     return this;
   }
+
+  /**
+   * Create this quaternion rotate around an arbitrary axis.
+   * @param axis - The axis
+   * @param rad - The rotation angle in radians
+   * @returns This quaternion
+   */
+  rotateAxisAngle(axis: Vector3, rad: number): Quaternion {
```
Comment (gz65555) [inline packages/math/src/Quaternion.ts:708]: Add unit test for this function.

File: packages/math/src/Quaternion.ts [hunk 3]
```diff
@@ -9,6 +9,7 @@ import { Vector3 } from "./Vector3";
 export class Quaternion implements IClone {
   /** @internal */
   static readonly _tempVector3 = new Vector3();
+  static readonly _tempQuat1 = new Quaternion();
```
Comment (gz65555) [inline packages/math/src/Quaternion.ts:12]: add `/** @internal */` comment.

File: packages/math/src/Quaternion.ts [hunk 4]
```diff
@@ -697,4 +699,16 @@ export class Quaternion implements IClone {
     Quaternion.lerp(this, quat, t, this);
     return this;
   }
+
+  /**
+   * Create this quaternion rotate around an arbitrary axis.
```
Comment (gz65555) [inline packages/math/src/Quaternion.ts:704]: Do not use **create** in the comment of Instance method.

**Review by GuoLei1990 — APPROVED** [review]
+1

**Pull request merged**